meteor/client/ui/i18n.ts

```diff
@@ -2,49 +2,76 @@ import * as i18n from 'i18next'
 import * as Backend from 'i18next-xhr-backend'
 import * as LanguageDetector from 'i18next-browser-languagedetector'
 import { reactI18nextModule } from 'react-i18next'
+import { WithManagedTracker } from '../lib/reactiveData/reactiveDataHelper'
+import { PubSub } from '../../lib/api/pubsub'
+import { TranslationsBundles } from '../../lib/collections/TranslationsBundles'
 
-let i18nTranslator: i18n.TranslationFunction<any, object, string>
-const i18nInstance = i18n
-	.use(Backend)
-	.use(LanguageDetector)
-	.use(reactI18nextModule)
-	.init(
-		{
-			fallbackLng: {
-				nn: ['nb', 'en'],
-				default: ['en'],
-			},
+const i18nOptions = {
+	fallbackLng: {
+		nn: ['nb', 'en'],
+		default: ['en'],
+	},
 
-			// have a common namespace used around the full app
-			ns: ['translations'],
-			defaultNS: 'translations',
+	// have a common namespace used around the full app
+	ns: ['translations'],
+	defaultNS: 'translations',
 
-			debug: false,
-			joinArrays: '\n',
+	debug: false,
+	joinArrays: '\n',
 
-			whitelist: ['en', 'nb', 'nn', 'sv'],
+	whitelist: ['en', 'nb', 'nn', 'sv'],
 
-			keySeparator: '→',
-			nsSeparator: '⇒',
-			pluralSeparator: '⥤',
-			contextSeparator: '⥤',
+	keySeparator: '→',
+	nsSeparator: '⇒',
+	pluralSeparator: '⥤',
+	contextSeparator: '⥤',
 
-			interpolation: {
-				escapeValue: false, // not needed for react!!
-			},
+	interpolation: {
+		escapeValue: false, // not needed for react!!
+	},
 
-			react: {
-				wait: true,
-			},
-		},
-		(err, t) => {
-			if (err) {
-				console.error('Error initializing i18Next', err)
-			} else {
-				i18nTranslator = t
+	react: {
+		wait: true,
+	},
+}
+
+class I18nContainer extends WithManagedTracker {
+	i18nInstance: i18n.i18n
+	i18nTranslator: i18n.TranslationFunction<any, object, string>
+
+	constructor() {
+		super()
+
+		this.i18nInstance = i18n
+			.use(Backend)
+			.use(LanguageDetector)
+			.use(reactI18nextModule)
+			.init(i18nOptions, (err, t) => {
+				if (err) {
+					console.error('Error initializing i18Next', err)
+				} else {
+					this.i18nTranslator = t
+				}
+			})
+
+		this.subscribe(PubSub.translationsBundles, null)
```

Comment:
Considering the possible amount of data this may load (if we have a lot of various translations), it might be a good idea to narrow this subscription down. If my memory serves me right, Meteor publications use up some memory per connection/subscription depending on the size of that subscribed data set. So it's a good idea to keep the subscriptions to a minimum. Perhaps this could be integrated as an additional i18next-backend so that only the locales requested by i18next (based on the used detection scheme) would be subscribed to? Just a thought.

Reply:
I'm not entirely sure of what constitutes as a large amount of data in this context, but the complete set of translations for core right now (which is the two variants of Norwegian) comes in at <90 kB of JSON data.
I'd expect the translations coming from blueprints to be significantly less than that, given that most of the text content should be in core. But I could be wrong :)